### PR TITLE
Allow interface specification in public ports

### PR DESCRIPTION
**Pull request by jhuie-zenoss**

### CODE REVIEW

File: web/ui/src/Services/ServiceDetailsController.js [hunk 1]
```diff
@@ -180,9 +180,10 @@
                 return resourcesFactory.addVHost(serviceId, serviceEndpoint, name);
             } else if(newPublicEndpoint.type === "port"){
                 var port = newPublicEndpoint.port;
+                var ip = newPublicEndpoint.ip;
```
Comment (rDr4g0n) [inline web/ui/src/Services/ServiceDetailsController.js]: where does this `ip` value come from?

File: web/ui/src/Services/ServiceDetailsController.js [hunk 2]
```diff
@@ -297,10 +298,13 @@
...
-                var host = $scope.defaultHostAlias;
-                // Port public endpoint port listeners are always on http
-                return "http://" + host + ":" + publicEndpoint.PortNumber;
+                if(publicEndpoint.PortNumber.startsWith(":")){
+                    var host = $scope.defaultHostAlias;
+                    // Port public endpoint port listeners are always on http
+                    return "http://" + host + publicEndpoint.PortNumber;
+                }else{
+                    return "http://" + publicEndpoint.PortNumber;
+                }
```
Comment (rDr4g0n) [inline web/ui/src/Services/ServiceDetailsController.js]: This code is looking for `PortNumber` property, but I think that property was renamed to `PortAddr`?

File: web/ui/src/Services/servicesFactory.js [hunk 3]
```diff
@@ -519,7 +519,7 @@
                         if(endpoint.PortList){
                             endpoint.PortList.forEach(function(port){
                                 acc.push({
-                                    PortNumber: port.PortNumber,
+                                    PortNumber: port.PortAddr,
```
Comment (rDr4g0n) [inline web/ui/src/Services/servicesFactory.js]: Rename `PortNumber` for consistency with the backend.

**Comment by rDr4g0n** [comment]
Also, since the port field has changed to "Port Address", i think it might be useful to include some placeholder text in the input that clarifies that it expects ip:port type format.

File: container/endpoint.go [hunk 4]
```diff
@@ -210,9 +210,9 @@ func buildExportedEndpoints(conn coordclient.Connection, tenantID string, state
 				}
 			}
 			if len(defep.PortList) > 0 {
-				exp.ports = []uint16{}
+				exp.ports = []string{}
```
Comment (smousa) [inline container/endpoint.go]: ports is not a good name for this attribute if it doesn't refer to ports anymore

**Pull request merged**